src/js/domManipulation.js

```diff
@@ -0,0 +1,186 @@
+const profileData = JSON.parse(localStorage.getItem("uploadProfile")) || { name: "Daniel Haribo" , image: "https://images.unsplash.com/photo-1687360440984-3a0d7cfde903?ixlib=rb-4.0.3&ixid=M3wxMjA3fDF8MHxlZGl0b3JpYWwtZmVlZHw0Nnx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=400&q=60"};
+const artistCardList = document.querySelector('.artist-card-list');
+const playListCardList = document.querySelector('.playlist-card-list');
+const recentPlayList = document.querySelector('.recent-playlist-list');
+const artistList = document.querySelector('.artist-list');
+const profileModal = document.getElementById("profile-modal");
+const displayUsername = document.getElementById("artist-card-greet");
+const profileNameElement = document.getElementById("user-profile-name");
+const profilePictureElement = document.getElementById("user-profile-picture");
+const profileNameInput = document.getElementById("upload-profile-name");
+const previewProfilePic = document.getElementById("upload-profile-pic");
+const favoriteList = document.getElementById('favorite-list');
+
+
+const getData = async () => {
+    const url = "https://raw.githubusercontent.com/Dina-petrick/musicData/main/db.json";
+    const response = await fetch(url);
+    const data = await response.json();
+    localStorage.setItem('fetchedData', JSON.stringify(data));
+}
+
+const createArtistCard = (data, index) => {
+    const artistCardItem = document.createElement("li");
+    artistCardItem.className = "artist-card-item";
+    artistCardItem.setAttribute("data-index", index);
+    artistCardItem.innerHTML = `
+        <div class="artist-card-info">
+            <img class="artist-card-img" src="${data.artistImage}" alt="artist">
+            <div class="artist-card-details">
+            <h4 class="fw-600 fs-20">${data.title}</h4>
+            <small class="fw-400 fs-14">${data.artist}</small>
+        </div>
+        <span class="artist-card-stats">${index + 1}</span>
+        </div>
+        <div class="artist-card-heart-wrap">
+            <span class="artist-card-heart fav-artist"><i class="fa-regular fa-heart heart-icon"></i></span>
+        </div>
+    `;
+    return artistCardItem;
+}
+
+const artistCard = (dataList) => {
+    dataList.forEach((data, index) => {
+        const artistCardItem = createArtistCard(data, index);
+        artistCardList.appendChild(artistCardItem);
+    })
+}
```

Comment:
or else you can make a generic loop function that will take in a `createComponent` function and give the html output